components/Layout.tsx

```diff
@@ -1,15 +1,35 @@
-import React from 'react'
+import React, {useEffect} from 'react'
 import Header from './Header/Header'
+import {useUser} from "@auth0/nextjs-auth0";
+import {userProfileState} from "../atoms/userProfile";
+import {useSetRecoilState} from "recoil";
+import {getUserFromFirebase} from "../lib/userHelper";
 
 interface LayoutProps {
     children: React.ReactNode
 }
 
-const Layout = ({ children }: LayoutProps) => {
+
+const Layout = ({children}: LayoutProps) => {
+    // useUser is used to get the current Auth0 user.
+    const {user, isLoading} = useUser();
```

Comment:
The default [Auth0 userProfile struct](https://auth0.github.io/nextjs-auth0/interfaces/frontend_use_user.userprofile.html) already stores a fair amount of user profile data. Do we think it's sufficient for our purposes? @FedericoCampe8 

If so, we wouldn't  have to create a separate profile collection for users.

Reply:
Good point, I thought so too at first then I see that we also store things like blocked users, posts list, etc., inside the user collection which is != from profiles (profile is just simple identification data). Maybe we can call it differently, not users but something like usersInfo or usersSemanticData or similar.